### DISCUSSION TITLE
How to run Bevy on Github Codespaces

### DISCUSSION THREAD

**Discussion by Deaths-Door** [bevyengine/bevy#14072] [Q&A] [answered]
After setting up my development environment on GitHub Codespaces and installing all the necessary libraries, I'm encountering a runtime error with bevy `0.13.2`

```
thread 'main' panicked at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_winit-0.13.2/src/lib.rs:138:14:
Failed to build event loop: Os(OsError { line: 786, file: "/root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.15/src/platform_impl/linux/mod.rs", error: Misc("neither WAYLAND_DISPLAY nor WAYLAND_SOCKET nor DISPLAY is set.") })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Then I set the` DISPLAY` environment variable (as suggested online, using `export DISPLAY=:0`) resulted in a different error message.

```
thread 'main' panicked at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_winit-0.13.2/src/lib.rs:138:14:
Failed to build event loop: NotSupported(NotSupportedError)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
````

The code that I am using is 
```rust
use bevy::prelude::*;

fn main() {
  App::new()
    .add_plugins(DefaultPlugins)
    .add_systems(Startup, setup)
    .run();
}

fn setup(mut commands: Commands) {
  commands.spawn(Camera2dBundle::default());
  commands.spawn(SpriteBundle {
    sprite: Sprite {
      color: Color::hex("98c379").unwrap(),
      custom_size: Some(Vec2::new(32.0, 32.0)),
      ..Default::default()
    },
    ..Default::default()
  });
  commands.spawn(SpriteBundle::default());
}
```

Is there a way to fix this error and successfully run my Bevy application on GitHub Codespaces?

**Comment by midstreeeam** [comment]
Try use `MinimalPlugins` rather than `DefaultPlugins`, which will call WinitPlugin.

**Reply by Deaths-Door** [reply]
Changing to `MinimalPlugins` causes it to run , however no window seems to be created
```
root@codespaces-015f06:/workspaces/coded-crusaders# cargo run
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.29s
     Running `target/debug/coded-crusaders`
```

**Comment by mockersf** [comment, marked as answer]
If you want to run rendering in codespaces, you'll need to setup a virtual x11 server, then connect with VNC or something similar. it will also be very slow as there isn't a GPU available on codespaces, even when paying, so you'll also need to install a software renderer.

It's generally a bad idea.

**Reply by Deaths-Door** [reply]
Oh ok, in that case I will continue trying out bevy on my local machine
